libparsec/crates/protocol/tests/authenticated_cmds/v4/shamir_recovery_setup.rs

```diff
@@ -0,0 +1,236 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+#![allow(clippy::unwrap_used)]
+
+use libparsec_protocol::authenticated_cmds::v4::shamir_recovery_setup::ShamirRecoverySetup;
+use libparsec_tests_lite::prelude::*;
+use libparsec_types::{DateTime, ShamirRevealToken};
+
+use super::authenticated_cmds;
+
+macro_rules! test_de_serialization {
+    ($encoded:literal, $expected:path, $load:path) => {
+        let raw = hex!($encoded);
+        let data = $load(&raw).unwrap();
+        p_assert_eq!(
+            data,
+            $expected,
+            "Expected hex is:\n\"{}\"",
+            encode_and_format_with_70_width($expected.dump().unwrap())
+        );
+
+        // Also test serialization round trip
+        let raw2 = data.dump().unwrap();
+        let data2 = $load(&raw2).unwrap();
+        p_assert_eq!(data2, $expected);
+    };
+}
+
+fn encode_and_format_with_70_width<T: AsRef<[u8]>>(data: T) -> String {
+    hex::encode(data)
+        .as_bytes()
+        .chunks(70)
+        .map(std::str::from_utf8)
+        .collect::<Result<Vec<&str>, _>>()
+        .unwrap()
+        .join(" \\\n")
+}
+
+pub fn rep_shamir_setup_already_exists() {
+    // Generated from Rust implementation (Parsec v3.0.0-b.6+dev 2024-06-06)
+    // Content:
+    //   "status": "shamir_setup_already_exist"
+    test_de_serialization!(
+        "81a6737461747573bb7368616d69725f73657475705f616c72656164795f6578697374 \
+73",
```

Comment:
```suggestion
        "81a6737461747573bb7368616d69725f73657475705f616c72656164795f6578697374"
        "73",
```

Invalid per: https://github.com/Scille/parsec-cloud/blob/master/docs/rfcs/1009-hexstring-format.md

Reply:
The strings are 70 col wide. The issue I have is that the kind of input accepted by the hex macro is not the way multi line strings are formatted. So I can try to make the test_de_serialization macro take the same kind of input as hex! for $encoded, but I'm not sure it's worth it. What do you think ?

The goal of test_de_serialization is that when you use it and the test fails, it outputs you the expected hex nicely broken down.